adlfs/tests/test_spec.py

```diff
@@ -2045,3 +2047,100 @@ def test_open_file_x(storage: azure.storage.blob.BlobServiceClient, tmpdir):
         with fs.open("data/afile", "xb") as f:
             pass
     assert fs.cat_file("data/afile") == b"data"
+
+
+def test_user_agent_blob_file_connection_str(
+    storage: azure.storage.blob.BlobServiceClient, mocker
+):
+    from adlfs import __version__
+
+    fs = AzureBlobFileSystem(
+        account_name=storage.account_name, connection_string=CONN_STR
+    )
+    mock_client = mocker.patch.object(
+        AIOBlobServiceClient, "from_connection_string", return_value=mocker.MagicMock()
+    )
+    mock_client.return_value.get_container_client = mocker.MagicMock()
+
+    f = AzureBlobFile(fs, "data/root/a/file.txt", mode="rb")
+    f.container_name = "data"
+    f.connect_client()
+
+    mock_client.assert_called_once()
+    assert "user_agent" in mock_client.call_args.kwargs
+    assert mock_client.call_args.kwargs["user_agent"] == f"adlfs/{__version__}"
```

Comment:
It would probably make sense to consolidate these assertions to a helper function (e.g., `assert_sets_adlfs_user_agent`) and we can just provide it the mock client method whether it be the mock initializer or the `from_connection_string` mock.